js/piwik.js

```diff
@@ -7634,6 +7645,159 @@ if (typeof window.Piwik !== 'object') {
                 return !configHasConsent;
             };
 
+            /**
+             * Returns a unique ID for this tracker instance.
+             *
+             * @returns {number}
+             */
+            this.getUniqueTrackerId = function () {
+                return this.uniqueTrackerId;
+            };
+
+            /**
+             * Loads extra tracker config via an HTTP request. Uses jsonp.
+             *
+             * @param {Tracker} tracker
+             * @param {Function} callback @see getExtraConfig
+             */
+            function loadExtraConfig(tracker, callback) {
+                extraTrackerConfigCallbacks.push(callback);
+
+                if (isAlreadyFetchingExtraConfig) {
```

Comment:
Not sure if important but do we maybe need to wait to make sure that `tracker.getPiwikUrl` and `tracker.getIdSite()` both return a value and are configured? I see there was some code around this in Heatmaps... 

Eg there could be an edge case when user configures the tracker manually, that maybe no Matomo URL or idSite is configured just yet when executing this logic (Heatmaps executes this logic on various occasions like onDomReady and onLoad). Eg we recommend to first only set a tracker URL and then at a later point the idSite see eg https://github.com/matomo-org/tag-manager/commit/f0e368d4e58ba9af86551fad4b7b578a5bb9761c#diff-45027e0449eb2627a9d7ae2d953aa2ef . If there was some time between these two events, it be technically possible that a plugin asks to load the config in that moment. Practically, it is rather unlikely though that this logic would be executed after a tracker URL was set and before setting an ID since JS wouldn't execute an event in between those lines... It could happen though depending how a user sets the idSite.

Also the flow could be like this:
* new Tracker('trackerUrl')
* A plugin is listening to the `TrackerSetup` event
* During that event a plugin requests the config to have it available as soon as possible
* The idSite might not be set yet

I reckon in this case it be a false implementation of the tracker plugin as it shouldn't do that. 

Probably my comment is confusing and I don't think we need it just yet maybe unless it's easy to implement in which case it would help with edge cases maybe.

I suppose it could be for the idSite eg like  this

```js
this.setIdSite = function (idSite) {
    configIdSite = idSite;
    if (extraTrackerConfigCallbacks.length > 0 && ! isAlreadyFetchingExtraConfig) {
        loadExtraConfig(this);
    }
}
```

Reply:
I think I fixed this case, can you take a look?